### PR TITLE
[SYCL] Fix name collisions in SYCL enums

### PR DESCRIPTION
**Pull request by denis-kabanov**
Correcting the definition of some macros causes name collisions if they are defined prior to including of CL/sycl.hpp.

### CODE REVIEW

File: sycl/include/CL/sycl/detail/cg.hpp [hunk 1]
```diff
@@ -150,22 +150,22 @@ class CG {
...
-    FILL = 7,
-    UPDATE_HOST = 8,
-    RUN_ON_HOST_INTEL = 9,
-    COPY_USM = 10,
-    FILL_USM = 11,
-    PREFETCH_USM = 12,
-    CODEPLAY_INTEROP_TASK = 13,
-    CODEPLAY_HOST_TASK = 14,
-    ADVISE_USM = 15,
+    __SYCL_NONE = 0,
```
Comment (bader) [inline sycl/include/CL/sycl/detail/cg.hpp:153]:
```suggestion
    None = 0,
```
?

Comment (bader) [reply sycl/include/CL/sycl/detail/cg.hpp:153]: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Renum-caps

Comment (denis-kabanov) [reply sycl/include/CL/sycl/detail/cg.hpp:153]: Hi, are you suggesting to change uppercase to lowercase (for example, __SYCL_NONE --> None)?

Comment (bader) [reply sycl/include/CL/sycl/detail/cg.hpp:153]: Yes. It's more readable and solves the collision problem.

Comment (bader) [reply sycl/include/CL/sycl/detail/cg.hpp:153]: It's called CamelCase.

Comment (bader) [reply sycl/include/CL/sycl/detail/cg.hpp:153]: ADVISE_USM --> AdviseUSM

Comment (bader) [reply sycl/include/CL/sycl/detail/cg.hpp:153]: To be more specific, we are following LLVM's coding standards. 
Here is relevant section - https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly

Comment (denis-kabanov) [reply sycl/include/CL/sycl/detail/cg.hpp]: fixed

File: sycl/include/CL/sycl/detail/cg.hpp [hunk 2]
```diff
@@ -150,22 +150,22 @@ class CG {
 
   /// Type of the command group.
   enum CGTYPE : unsigned int {
```
Comment (bader) [inline sycl/include/CL/sycl/detail/cg.hpp:152]: What about enumeration name? CGTYPE -> CGType?

File: sycl/test/regression/check_simple_name_collisions.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
```
Comment (bader) [inline sycl/test/regression/check_simple_name_collisions.cpp:1]:
```suggestion
// RUN: %clangxx -fsycl -fsyntax-only -Xclang -verify %s
// expected-no-diagnostics
```

**Review by bader — DISMISSED** [review]
LGTM, just a couple of other suggetions.

File: sycl/test/regression/check_simple_name_collisions.cpp (hunk 3 above)
Comment (denis-kabanov) [reply sycl/test/regression/check_simple_name_collisions.cpp]: Done

File: sycl/include/CL/sycl/detail/cg.hpp (hunk 2 above)
Comment (denis-kabanov) [reply sycl/include/CL/sycl/detail/cg.hpp]: Done

**Review by bader — APPROVED** [review]
LGTM, although I think someone from runtime team should take a look and make sure this change doesn't break library ABI.

**Comment by dm-vodopyanov** [comment]
ABI didn't break, luckily, we don't have such enums in exported symbols.
@sergey-semenov, can you please review?

**Review by bader — APPROVED** [review]

**Review by sergey-semenov — APPROVED** [review]

**Comment by bader** [comment]
Next, I would expand `CG` to `CommandGroup` everywhere.

**Comment by denis-kabanov** [comment]
> Next, I would expand `CG` to `CommandGroup` everywhere.

Created an internal ticket

**Pull request merged**